python/rapidsmpf/rapidsmpf/shuffler.pyx

```diff
@@ -147,25 +157,66 @@ cdef class Shuffler:
         with nogil:
             deref(self._handle).insert(move(_chunks))
 
-    def insert_finished(self, uint32_t pid):
+    def concat_insert(self, chunks: Mapping[int, PackedData]):
         """
-        Mark a partition as finished.
+        Insert a batch of packed (serialized) chunks into the shuffle while
+        concatenating the chunks based on the destination rank.
 
-        This informs the shuffler that no more chunks for the specified partition
+        Parameters
+        ----------
+        chunks
+            A map where keys are partition IDs (``int``) and values are packed
+            data (``PackedData``).
+
+        Notes
+        -----
+        This method adds the given chunks to the shuffle, associating them with their
+        respective partition IDs.
+
+        There are some considerations for using this method:
+        - The chunks are grouped by the destination rank of the partition ID and
+        concatenated on device memory.
+        - Caller thread will perform the concatenation, and hence it will be blocked.
```

Comment:
Will the caller thread "be blocked", or will *it* block while doing the concatenation (like the actual set of memcpy operations needed to do the concat)?

```suggestion
        - The caller thread will perform the concatenation, and hence it will be blocked.
```

Reply:
Caller thread will be doing the actual memory copies.